src/Infrastructure.EntityFramework/Configurations/TransactionEntityTypeConfiguration.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Bit.Infrastructure.EntityFramework.Models;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Bit.Infrastructure.EntityFramework.Configurations;
+
+public class TransactionEntityTypeConfiguration : IEntityTypeConfiguration<Transaction>
+{
+    public void Configure(EntityTypeBuilder<Transaction> builder)
+    {
+        builder
+            .Property(t => t.Id)
+            .ValueGeneratedNever();
+
+        builder
+            .HasIndex(t => new { t.Gateway, t.GatewayId })
+            .HasFilter($"{nameof(Transaction.Gateway)} IS NOT NULL AND {nameof(Transaction.GatewayId)} IS NOT NULL")
```

Comment:
Unless, there is a more EF way to express this filter, I think I might veto it. Since that is probably just pasted somewhere it is likely that it might not work for all providers. Plus, I would have to check on this more, but I think the Transaction table is only used for cloud so it's not a huge loss if it doesn't have all the same indexes.

Reply:
Unfortunately, this is the [EF way](https://learn.microsoft.com/en-us/ef/core/modeling/indexes?tabs=fluent-api#index-filter) from what I saw. 

Do you think there is any value in having the index without the filter?

Reply:
Yeah it looks like this is the only way they offer, I was hoping for something like: `t => t.Gateway != null && t.GatewayId != null` and then was hoping it would translate correctly for each provider but it doesn't look like that is possible right now.

For these tables that, I believe, are cloud only. I would say lets go with not index at all. But I also see that you have other ones like `Organization.Identifier`. We for sure can't do the index without the filter. It's quite likely for an organization identifier to exist without an identifier so it could cause issues without the filter, but we also do want to enforce uniqueness of non-null ones. Now I don't know how common it would be to have more than one organization that has any identifier on SH possibly that number is 0. 

The OrganizationSponsorships one is the most difficult one. It is used on SH and could have as many rows as there are users in the org. We may have to hand write that filter for each provider.